magenta/lib/melodies_lib.py

```diff
@@ -248,6 +254,10 @@ def _get_last_on_off_events(self):
         return (i, last_off)
     raise ValueError('No events in the stream')
 
+  def copy(self):
+    """Returns a deep copy of this MonphonicMelody object."""
```

Comment:
Yeah I went back and forth on this but what we really need is a deep copy
so that we aren't accidentally modifying the state of the original object.
Should we just name the functions deepcopy?

On Sep 1, 2016 9:42 PM, "cghawthorne" <notifications@github.com> wrote:

> In magenta/lib/melodies_lib.py
> <https://github.com/tensorflow/magenta/pull/193#discussion_r77292401>:
>
> > @@ -248,6 +254,10 @@ def _get_last_on_off_events(self):
> >          return (i, last_off)
> >      raise ValueError('No events in the stream')
> >
> > +  def copy(self):
> > +    """Returns a deep copy of this MonphonicMelody object."""
>
> nit: probably shouldn't call this a "deep" copy because it uses the
> shallow copy operator (*copy*). I think in practice it does end up being
> a deep copy, but that's only because from_event_list happens to use an
> iterator over a list of ints.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/magenta/pull/193/files/0fc3768aff533f8c5f2143f3a445d279144410a8#r77292401>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCa6Nlu3MJ8Vf6Bav4CUzfP6_9u3CcIks5ql6kkgaJpZM4JzUc5>
> .
>